src/profile/v100/proto_profile_maybe_with_legacy_p2p_links.rs

```diff
@@ -0,0 +1,124 @@
+use crate::prelude::*;
+
+#[derive(Deserialize, Serialize, Eq, PartialEq, Debug)]
+pub struct ProtoProfileMaybeWithLegacyP2PLinks {
+    #[serde(rename = "appPreferences")]
+    pub app_preferences: ProtoAppPreferencesMaybeWithLegacyP2PLinks,
+}
+
+#[derive(Deserialize, Serialize, Eq, PartialEq, Debug)]
+pub struct ProtoAppPreferencesMaybeWithLegacyP2PLinks {
+    #[serde(rename = "p2pLinks")]
+    pub p2p_links: Vec<ProtoDummyLinkMaybeWithLegacyP2PLinks>,
+}
+
+#[derive(Deserialize, Serialize, Eq, PartialEq, Debug)]
+pub struct ProtoDummyLinkMaybeWithLegacyP2PLinks {
+    #[serde(rename = "displayName")]
+    pub display_name: String,
+}
+
+impl HasSampleValues for ProtoDummyLinkMaybeWithLegacyP2PLinks {
+    fn sample() -> Self {
+        Self {
+            display_name: String::from("Sample Display Name"),
+        }
+    }
+
+    fn sample_other() -> Self {
+        Self {
+            display_name: String::from("Sample Display Name Other"),
+        }
+    }
+}
+
+impl HasSampleValues for ProtoAppPreferencesMaybeWithLegacyP2PLinks {
+    fn sample() -> Self {
+        Self {
+            p2p_links: Vec::from([
+                ProtoDummyLinkMaybeWithLegacyP2PLinks::sample(),
+            ]),
+        }
+    }
+
+    fn sample_other() -> Self {
+        Self {
+            p2p_links: Vec::from([
+                ProtoDummyLinkMaybeWithLegacyP2PLinks::sample_other(),
+            ]),
+        }
+    }
+}
+
+impl HasSampleValues for ProtoProfileMaybeWithLegacyP2PLinks {
+    fn sample() -> Self {
+        Self {
+            app_preferences: ProtoAppPreferencesMaybeWithLegacyP2PLinks::sample(
+            ),
```

Comment:
strange newline in middle of parenthesis, strange that cargo fmt did not complain? or maybe IT produced this :P

Reply:
cargo fmt produced it :(

Reply:
damn :D do a fmt ignore then :)

Reply:
https://stackoverflow.com/a/67289474